packages/react/src/Button/IconButton.tsx

```diff
@@ -4,20 +4,75 @@ import type {ForwardRefComponent as PolymorphicForwardRefComponent} from '../uti
 import {ButtonBase} from './ButtonBase'
 import {defaultSxProp} from '../utils/defaultSxProp'
 import {generateCustomSxProp} from './Button'
+import {TooltipContext, Tooltip} from '../TooltipV2/Tooltip'
+import {TooltipContext as TooltipContextV1} from '../Tooltip/Tooltip'
 
-const IconButton = forwardRef(({sx: sxProp = defaultSxProp, icon: Icon, ...props}, forwardedRef): JSX.Element => {
-  let sxStyles = sxProp
-  // grap the button props that have associated data attributes in the styles
-  const {size} = props
+const IconButton = forwardRef(
+  (
+    {
+      sx: sxProp = defaultSxProp,
+      icon: Icon,
+      'aria-label': ariaLabel,
+      description,
+      disabled,
+      tooltipDirection,
+      // This is planned to be a temporary prop until the default tooltip on icon buttons are fully rolled out.
+      disableTooltip = false,
+      ...props
+    },
+    forwardedRef,
+  ): JSX.Element => {
+    let sxStyles = sxProp
+    // grap the button props that have associated data attributes in the styles
+    const {size} = props
 
-  if (sxProp !== null && Object.keys(sxProp).length > 0) {
-    sxStyles = generateCustomSxProp({size}, sxProp)
-  }
+    if (sxProp !== null && Object.keys(sxProp).length > 0) {
+      sxStyles = generateCustomSxProp({size}, sxProp)
+    }
 
-  return (
-    // @ts-expect-error StyledButton wants both Anchor and Button refs
-    <ButtonBase icon={Icon} data-component="IconButton" sx={sxStyles} type="button" {...props} ref={forwardedRef} />
-  )
-}) as PolymorphicForwardRefComponent<'button' | 'a', IconButtonProps>
+    // If the icon button is already wrapped in a tooltip, do not add one.
+    const {tooltipId} = React.useContext(TooltipContext) // Tooltip v2
+    const {tooltipId: tooltipIdV1} = React.useContext(TooltipContextV1) // Tooltip v1
+
+    const hasExternalTooltip = tooltipId || tooltipIdV1
+    const withoutTooltip =
+      disableTooltip || disabled || ariaLabel === undefined || ariaLabel === '' || hasExternalTooltip
```

Comment:
(non blocking question) is the check for empty/undefined aria-label temporary for smooth adoption or a permanent check?

Reply:
Great question! There is a possibility of `aria-label` being undefined i.e. when `aria-labelledby` is provided. 

When an icon button has an `aria-labelledby` prop, very high likely that it will have a tooltip ([search results to reference](https://github.com/search?q=%3CIconButton+AND+aria-labelledby+AND+repo%3Agithub%2Fgithub&type=code)) so I chose to not complicate things and not to show a tooltip when there is an `aria-labelledby`. I guess there is a still a change for cases like below to show up but imo it is not a traditional icon button API and we might not need to cover this edge case? They can always wrap it with tooltip for additional text. Let me know what you think 🙌🏻 


```jsx
<IconButton aria-labelledby="some-element-to-label" description="additional text" .../>
```

Reply:
Sounds fair to me, up to you!